src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -112,189 +73,215 @@ public static unsafe bool ListChildPids(int ppid, out int[] buffer)
-            if (snapshot == IntPtr.Zero)
+            var children = new List<Process>();
+            if (libproc.ListChildPids(process.Id, out int[] childPids))
             {
-                child = null;
-                return false;
+                foreach (var childPid in childPids)
+                {
+                    children.Add(Process.GetProcessById(childPid));
+                }
             }
 
-            try
-            {
-                int ppid = process.Id;
+            return children;
+        }
+    }
 
-                var processEntry = new Kernel32.ProcessEntry32W { Size = sizeof(Kernel32.ProcessEntry32W) };
+    public class CoreclrTestWrapperLib
+    {
+        public const int EXIT_SUCCESS_CODE = 0;
+        public const string TIMEOUT_ENVIRONMENT_VAR = "__TestTimeout";
+        
+        // Default timeout set to 10 minutes
+        public const int DEFAULT_TIMEOUT_MS = 1000 * 60 * 10;
 
-                bool success = Kernel32.Process32FirstW(snapshot, ref processEntry);
-                while (success)
-                {
-                    if (processEntry.ParentProcessID == ppid)
-                    {
-                        try
-                        {
-                            Process c = Process.GetProcessById(processEntry.ProcessID);
-                            if (c.ProcessName.Equals(childName, StringComparison.OrdinalIgnoreCase))
-                            {
-                                child = c;
-                                return true;
-                            }
-                            c.Dispose();
-                        }
-                        catch {}
-                    }
+        public const string COLLECT_DUMPS_ENVIRONMENT_VAR = "__CollectDumps";
+        public const string CRASH_DUMP_FOLDER_ENVIRONMENT_VAR = "__CrashDumpFolder";
 
-                    success = Kernel32.Process32NextW(snapshot, ref processEntry);
-                }
+        static bool CollectCrashDump(Process process, string path)
+        {
+            ProcessStartInfo createdumpInfo = null;
+            string coreRoot = Environment.GetEnvironmentVariable("CORE_ROOT");
+            string createdumpPath = Path.Combine(coreRoot, "createdump");
+            string arguments = $"--name \"{path}\" {process.Id} --withheap";
+            Process createdump = new Process();
 
-                child = null;
-                return false;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                createdump.StartInfo.FileName = createdumpPath + ".exe";
+                createdump.StartInfo.Arguments = arguments;
             }
-            finally
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
             {
-                Kernel32.CloseHandle(snapshot);
+                createdump.StartInfo.FileName = "sudo";
```

Comment:
Is sudo absolutely required here? I believe this currently just "happens" to work because we have root access in the linux containers

Reply:
Keep in mind that our containers are also not explicitly required to have sudo installed as well, they may just run as su, because of this it seems fragile to use it in this way.

Reply:
Also I understand that this was existing code. This is mostly a question going forward

Reply:
I'm not sure for Linux.  If we are already running as `root` I don't think it is necessary.  As for macOS, however, I think the permissions in 10.15+ require `sudo` for createdump to work.  @mikem8361 ?